animation_nodes/nodes/vector/c_utils.pyx

```diff
@@ -105,3 +105,11 @@ def offset3DVectors(Vector3DList vectors, VirtualVector3DList offsets, FloatList
         vectors.data[i].x += offset.x * influence
         vectors.data[i].y += offset.y * influence
         vectors.data[i].z += offset.z * influence
+
+def calculateVectorDotProducts(VirtualVector3DList vectors1, VirtualVector3DList vectors2):
+    cdef Py_ssize_t i
+    cdef Py_ssize_t amount = VirtualVector3DList.getMaxRealLength(vectors1, vectors2)
```

Comment:
Compute the amount in the python code and pass it.

Reply:
Done

Reply:
I have seen many times we compute amount outside. Is there any reason for that?